crates/core_simd/src/pointers.rs

```diff
@@ -0,0 +1,116 @@
+                self.0.eq(&other.0)
+            }
+        }
+
+        impl<T> core::cmp::Eq for $name<T> {}
+
+        impl<T> core::cmp::PartialOrd for $name<T> {
+            #[inline]
+            fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+                self.0.partial_cmp(&other.0)
+            }
+        }
+
+        impl<T> core::cmp::Ord for $name<T> {
+            fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+                self.0.cmp(&other.0)
+            }
+        }
+
+        impl<T> $name<T> {
+            /// Construct a vector by setting all lanes to the given value.
+            #[inline]
+            pub fn splat(value: *$mut T) -> Self {
+                Self(<$underlying>::splat(value as isize), PhantomData)
+            }
+            call_counting_args! { $lanes => define_pointer_vector => new $underlying | *$mut T | }
+        }
+
+        // array references
+        impl<T> AsRef<[*$mut T; $lanes]> for $name<T> {
+            #[inline]
+            fn as_ref(&self) -> &[*$mut T; $lanes] {
+                unsafe { &*(self as *const _ as *const _) }
+            }
+        }
+
+        impl<T> AsMut<[*$mut T; $lanes]> for $name<T> {
+            #[inline]
+            fn as_mut(&mut self) -> &mut [*$mut T; $lanes] {
+                unsafe { &mut *(self as *mut _ as *mut _) }
+            }
+        }
+
+        // slice references
+        impl<T> AsRef<[*$mut T]> for $name<T> {
+            #[inline]
+            fn as_ref(&self) -> &[*$mut T] {
+                AsRef::<[*$mut T; $lanes]>::as_ref(self)
+            }
+        }
+
+        impl<T> AsMut<[*$mut T]> for $name<T> {
+            #[inline]
+            fn as_mut(&mut self) -> &mut [*$mut T] {
+                AsMut::<[*$mut T; $lanes]>::as_mut(self)
+            }
+        }
+
+        // splat
+        impl<T> From<*$mut T> for $name<T> {
+            #[inline]
+            fn from(value: *$mut T) -> Self {
+                Self::splat(value)
+            }
+        }
+    };
+    { new $underlying:ty | $type:ty | $($var:ident)* } => {
+        /// Construct a vector by setting each lane to the given values.
+        #[allow(clippy::too_many_arguments)]
+        #[inline]
+        pub fn new($($var: $type),*) -> Self {
+            Self(<$underlying>::new($($var as isize),*), PhantomData)
+        }
+    };
+}
+
+define_pointer_vector! { #[doc = "Vector of two mutable pointers"] mptrx2 => isizex2 => 2, mut }
+define_pointer_vector! { #[doc = "Vector of four mutable pointers"] mptrx4 => isizex4 => 4, mut }
+define_pointer_vector! { #[doc = "Vector of eight mutable pointers"] mptrx8 => isizex8 => 8, mut }
+define_pointer_vector! { #[doc = "Vector of two const pointers"] cptrx2 => isizex2 => 2, const }
+define_pointer_vector! { #[doc = "Vector of four const pointers"] cptrx4 => isizex4 => 4, const }
+define_pointer_vector! { #[doc = "Vector of eight const pointers"] cptrx8 => isizex8 => 8, const }
```

Comment:
What is exactly the purpose of pointer vectors?

Reply:
I've never done it, but there certainly may be a good reason to perform simultaneous ops on pointers (strided algorithms come to mind).  One thought (@Lokathor ?) is it actually worth providing both a `*const T` and `*mut T` vector?  `core::sync::atomic::AtomicPtr` only operates on `*mut T` and leaves it up to the user to ensure correct mutability.  I propose doing that instead, and just calling it `pointerxN`.

Reply:
`packed_simd` has them... but I think we can leave them out for now. Let's wait for someone to ask for it.

Because people can also use `usize` if they want.

I don't recall seeing any op at the 128-bit or 256-but level that's like, "take a vector of pointers and load each one into a vector of whatever"